### ISSUE TITLE
AutoCloseable not found in Robolectric 2.3

### ISSUE THREAD

**Issue by rnevet** [robolectric/robolectric#1002]
I'm using Robolectric and trying to test parsing a XML response. I'm getting the following exception:
```
java.lang.NoClassDefFoundError: java/lang/AutoCloseable
    at android.util.Xml.parse(Xml.java:80)
    at com.test.app.network.parser.soap.BaseSoapParser.doParse(BaseSoapParser.java:57)
    at com.test.app.network.parser.soap.RecipeParser.parse(RecipeParser.java:57)
    at com.test.app.network.command.RecipeCommand.searchRecipes(RecipeCommand.java:64)
    at com.test.app.data.PrepopulateDB.prepopulateDB(PrepopulateDB.java:53)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
    at org.robolectric.RobolectricTestRunner$2.evaluate(RobolectricTestRunner.java:230)
    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
    at org.robolectric.RobolectricTestRunner$1.evaluate(RobolectricTestRunner.java:172)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
    at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)
    at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
    at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:467)
    at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:683)
    at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:390)
    at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:197)
Caused by: java.lang.ClassNotFoundException: java.lang.AutoCloseable
    at java.net.URLClassLoader$1.run(URLClassLoader.java:202)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
    at org.robolectric.bytecode.AsmInstrumentingClassLoader.loadClass(AsmInstrumentingClassLoader.java:100)
    at android.util.Xml.$$robo$$Xml_e8aa_parse(Xml.java:80)
    at android.util.Xml.parse(Xml.java)
    at com.test.app.network.parser.soap.BaseSoapParser.doParse(BaseSoapParser.java:57)
    at com.test.test.app.network.parser.soap.RecipeParser.parse(RecipeParser.java:57)
    at com.test.app.network.command.RecipeCommand.searchRecipes(RecipeCommand.java:64)
    at com.test.app.data.PrepopulateDB.prepopulateDB(PrepopulateDB.java:53)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    ... 22 more
```

The thing is that "AutoCloseable" was only added in Java 7 and since Android uses Java 6 I can understand why the class isn't found - any help will be appreciated.

Libraries: robolectric-2.3-20140204.032758-112-jar-with-dependencies.jar android-all-4.4_r1-robolectric-0.jar

**Comment by rburgst** [comment]
@hkliya what is your solution?

**Comment by hkliya** [comment]
```
SAXParserFactory factory = SAXParserFactory.newInstance();
SAXParser parser = factory.newSAXParser();
XMLReader reader = parser.getXMLReader();
YourHandler handler = new YourHandler();
reader.setContentHandler(handler);
InputSource is = new InputSource(this.getClassLoader().getResourceAsStream("your.xml"));
reader.parse(is);
```

**Comment by grub-** [comment]
Very annoying issue same result on @Config(emulateSdk = 21).

@hkliya Thanks for workaround!

**Comment by erd** [comment]
The `AutoCloseable` issue should be solved in 3.0.
